bin/import_cv.py

```diff
@@ -9,101 +9,132 @@
 
 import csv
 import tarfile
+import requests
 import subprocess
 from glob import glob
 from os import path
 from sox import Transformer
 from threading import Lock
-from tensorflow.contrib.learn.python.learn.datasets import base
-from tensorflow.python.platform import gfile
 from multiprocessing.dummy import Pool
 from multiprocessing import cpu_count
 from util.progress import print_progress
 
-FIELDNAMES = ["wav_filename", "wav_filesize", "transcript"]
+FIELDNAMES = ['wav_filename', 'wav_filesize', 'transcript']
 SAMPLE_RATE = 16000
 MAX_SECS = 10
-CV_DIR = "cv_corpus_v1"
-CV_DATA = "cv_corpus_v1.tar.gz"
-CV_DATA_URL = "https://s3.us-east-2.amazonaws.com/common-voice-data-download/cv_corpus_v1.tar.gz"
+ARCHIVE_DIR_NAME = 'cv_corpus_v1'
+ARCHIVE_NAME = ARCHIVE_DIR_NAME + '.tar.gz'
+ARCHIVE_URL = 'https://s3.us-east-2.amazonaws.com/common-voice-data-download/' + ARCHIVE_NAME
 
-def _download_and_preprocess_data(data_dir):
+def _download_and_preprocess_data(target_dir):
     # Making path absolute
-    data_dir = path.abspath(data_dir)
+    target_dir = path.abspath(target_dir)
     # Conditionally download data
-    local_file = base.maybe_download(CV_DATA, data_dir, CV_DATA_URL)
+    archive_path = _maybe_download(ARCHIVE_NAME, target_dir, ARCHIVE_URL)
     # Conditionally extract common voice data
-    _maybe_extract(data_dir, CV_DIR, local_file)
+    _maybe_extract(target_dir, ARCHIVE_DIR_NAME, archive_path)
     # Conditionally convert common voice CSV files and mp3 data to DeepSpeech CSVs and wav
-    _maybe_convert_sets(data_dir, CV_DIR)
+    _maybe_convert_sets(target_dir, ARCHIVE_DIR_NAME)
 
-def _maybe_extract(data_dir, extracted_data, archive):
-    # If data_dir/extracted_data does not exist, extract archive in data_dir
-    if not gfile.Exists(path.join(data_dir, extracted_data)):
-        print('Extracting "%s"...' % archive)
-        with tarfile.open(archive) as tar:
+def _maybe_download(archive_name, target_dir, archive_url):
```

Comment:
Because of [this](https://github.com/mozilla/DeepSpeech/issues/982#issuecomment-346857514), which hides a problem in the temp-file copying in maybe_download. It occurs from time to time also on my end and seems to be related to the temporary file lifetime. The corpus is downloaded to a temp file first and then copied over to the target directory. The copy (erratically) failed on my end and results in the very same behavior as in the linked comment. 
There was also no progress indication and the temp-file design prevented resuming if the download got interrupted.

Reply:
Could you open up an issue for all the other importers that have this issue?